src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,41 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> dict:
+    assert HAVE_NUMBA, "spike_vector_to_dict() requires `numba`!"
+
+    spike_trains = _vector_to_dict(
+        spike_vector["sample_index"].astype(np.int64),
+        spike_vector["unit_index"].astype(np.int64),
+        spike_vector["segment_index"].astype(np.int64),
+    )
+
+    return spike_trains
+
+
+if HAVE_NUMBA:
+
+    @numba.jit((numba.int64[::1], numba.int64[::1], numba.int64[::1]), nopython=True, nogil=True, cache=True)
```

Comment:
Have you tested not having eager compilation? If you really want to use eager compilation (when I tested this myself in the past it yield no benefits) this should support all the integers types.
Let's not use cache as I am unsure how that plays out with multiprocessing and the numba documentation is full of caveasts.

Reply:
I haven't tried non-eager compilation (I just personally don't believe in not typing when compiling ^^')
I don't even know how it works to be honest, and the spike vector forces int64 anyway (look at `to_spike_vector`).
But feel free to change it if you prefer it another way!

For the cache, isn't it important?
I mean if you need to recompile it everytime you call it, then you lose a big portion of the time saved?

Reply:
Concerning typing: 
If you can have a more general function for no price in performance then that's better. I am fine with the typing, just add the more general version. We migt change some of the typing of the spike vector. The typing of the units and the segments is too large for what they represent for example. It is not good to not have this implicit coupling between different parts of the library (the implicit coupling here is the typing of the spike vector with the typing that this function takes).

Concerning cache:

Yes, caching will improve speed when calling multiple times. When profiling you should reflect the two cases. The first run of the function and the second. Users of the library will be exposed to the first use and that is important information.

I am concerned about cache invalidation and how this will play out with multiprocessing:
https://numba.readthedocs.io/en/stable/developer/caching.html#cache-clearing

This option works great when using it for your personal scripts when you have access to your system and know the code well so you can troubleshoot. But this is library code and we don't have control of what environments the users will run the code. Cache invalidation is a big problem and I don't know enough about how this works to take the risk.

I prefer to go with the safe option, then when we have more knowledge and ways of testing this we can enable caching globally for numba. This is better than having to solve difficult bug issues of users. Trade-off between maintanability and optimization.